File: spring-rest2ts-examples/web/src/app/_model/example/biz/TestUserManager.ts
//此代码为生成的代码请勿修改 !!!
import {HttpClient, HttpHeaders, HttpParams} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {TbCond} from '../../example/biz/TbCond'
import {TestUser3} from '../../example/biz/TestUser3'
import {UserProfileDTO} from '../../example/biz/UserProfileDTO'
import {ITableObj} from '../../example/biz/core/ITableObj'
import {MetaField} from '../../example/dev/meta/fields/MetaField'
import {AddressDTO} from '../../example/model/AddressDTO'
import {OrderDTO} from '../../example/model/OrderDTO'
import {PersonDTO} from '../../example/model/PersonDTO'
import {OrderPaymentStatus} from '../../example/model/enums/OrderPaymentStatus'
import {Observable} from 'rxjs';
import {map} from 'rxjs/operators';

@Injectable({providedIn: "root"})
export class TestUserManager implements ITableObj<OrderDTO, PersonDTO, AddressDTO, AddressDTO> {
      public static readonly h: string = "hhhh";
      public static readonly asda: string = "asda";
      public static readonly objs: TestUser3<number> [] = [{"t3":1},{"t3":1}];
      httpService: HttpClient;


 public constructor(httpService: HttpClient) {
    this.httpService = httpService;
 }

 public delete(keyIDs: number[]): Observable<void>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post<void>('/download/delete', keyIDs , {headers});
 }

 public getByID(keyID: number): Observable<PersonDTO>  {
    return this.httpService.post('/download/getByID/' + keyID + '', null , {responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getByOwn(req: {[key: string] :UserProfileDTO}): Observable<{[key: string] :UserProfileDTO}>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post('/download/getByOwn', JSON.stringify(req) , {headers, responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getByOwn2(req: {[key: string] :OrderPaymentStatus}): Observable<{[key: string] :OrderPaymentStatus}>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post('/download/getByOwn2', JSON.stringify(req) , {headers, responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getDetailFieldSchema(): Observable<MetaField[]>  {
    return this.httpService.post('/download/getDetailFieldSchema', null , {responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getInsertFieldSchema(): Observable<MetaField[]>  {
    return this.httpService.post('/download/getInsertFieldSchema', null , {responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getQueryCondFieldSchema(): Observable<MetaField[]>  {
    return this.httpService.post('/download/getQCondFieldSchema', null , {responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getQueryFieldSchema(): Observable<MetaField[]>  {
    return this.httpService.post('/download/getQueryFieldSchema', null , {responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public getTotal(conds: TbCond[]): Observable<number>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post<number>('/download/getTotal', conds , {headers});
 }

 public getUpdateFieldSchema(): Observable<MetaField[]>  {
    return this.httpService.post('/download/getUpdateFieldSchema', null , {responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public insert(insertObj: OrderDTO): Observable<void>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post<void>('/download/insert', insertObj , {headers});
 }

 public query(start: number, max: number, orderBy: string, bAsc: boolean, conds: TbCond[]): Observable<AddressDTO[]>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post('/download/query/' + start + '/' + max + '/' + orderBy + '/' + bAsc + '', JSON.stringify(conds) , {headers, responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public queryByID(keyID: number): Observable<AddressDTO>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post('/download/queryByID', JSON.stringify(keyID) , {headers, responseType: 'text'}).pipe(map(res => JSON.parse(res)));
 }

 public update(keyID: number, dataModel: PersonDTO): Observable<void>  {
    const headers = new HttpHeaders().set('Content-type', 'application/json');
    return this.httpService.post<void>('/download/update/' + keyID + '', dataModel , {headers});
 }

}

File: spring-rest2ts-examples/java/src/main/java/example/biz/TestUserManager.java
package example.biz;

import com.blueveery.springrest2ts.filter.GenTsField;
import example.biz.core.ITableObj;
import example.dev.meta.fields.MetaField;
import example.model.OrderDTO;
import example.model.AddressDTO;
import example.model.PersonDTO;
import com.blueveery.springrest2ts.filter.GenTsController;
import example.model.enums.OrderPaymentStatus;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Created by yangyi0 on 2021/1/26.
 */
@RestController
@GenTsController
@RequestMapping(value = "download")
public class TestUserManager implements ITableObj<OrderDTO, PersonDTO, AddressDTO, AddressDTO> {

    @GenTsField()
    public static final String h = "hhhh";
    @GenTsField()
    public static final String asda = "asda";

    @GenTsField()
    public static final List<TestUser3<Integer>> objs = new ArrayList<>();

    static {
        TestUser3<Integer> e = new TestUser3<>();
        e.setT3(1);
        objs.add(e);
        objs.add(e);
    }
    /**
     * 声明一个函数
     *
     * @param req
     * @return
     */
    @RequestMapping("/getByOwn")
    public Map<String,UserProfileDTO> getByOwn(@RequestBody @Validated Map<String,UserProfileDTO> req) {
        return req;
    }
    @RequestMapping("/getByOwn2")
    public Map<String, OrderPaymentStatus> getByOwn2(@RequestBody Map<String,OrderPaymentStatus> req) {
        return req;
    }

    @Override
    public AddressDTO queryByID(long keyID) throws Exception {
        return null;
    }

    @Override
    public List<MetaField> getDetailFieldSchema() {
        return null;
    }

    @Override
    public void delete(List<Long> keyIDs) throws Exception {

    }

    @Override
    public List<MetaField> getInsertFieldSchema() {
        return null;
    }

    @Override
    public void insert(OrderDTO insertObj) throws Exception {

    }

    @Override
    public List<MetaField> getQueryCondFieldSchema() {
        return null;
    }

    @Override
    public long getTotal(List<TbCond> conds) throws Exception {
        return 0;
    }

    @Override
    public List<MetaField> getQueryFieldSchema() {
        return null;
    }

    @Override
    public List<AddressDTO> query(long start, int max, String orderBy, boolean bAsc, List<TbCond> conds) throws Exception {
        return null;
    }

    @Override
    public void update(long keyID, PersonDTO dataModel) throws Exception {

    }

    @Override
    public List<MetaField> getUpdateFieldSchema() {
        return null;
    }

    @Override
    public PersonDTO getByID(long keyID) throws Exception {
        return null;
    }
}
